x/stablestake/keeper/debt.go

```diff
@@ -23,23 +23,24 @@ func (k Keeper) getDebt(ctx sdk.Context, addr sdk.AccAddress) (debt types.Debt)
 			BorrowTime:            uint64(ctx.BlockTime().Unix()),
 			LastInterestCalcTime:  uint64(ctx.BlockTime().Unix()),
 			LastInterestCalcBlock: uint64(ctx.BlockHeight()),
+			PoolId:                poolId,
 		}
 	}
 
 	k.cdc.MustUnmarshal(bz, &debt)
 	return
 }
 
-func (k Keeper) GetDebt(ctx sdk.Context, addr sdk.AccAddress) types.Debt {
-	debt := k.getDebt(ctx, addr)
-	debt.InterestStacked = debt.InterestStacked.Add(k.GetInterest(ctx, debt.LastInterestCalcBlock, debt.LastInterestCalcTime, debt.Borrowed.ToLegacyDec()))
+func (k Keeper) GetDebt(ctx sdk.Context, addr sdk.AccAddress, poolId uint64) types.Debt {
+	debt := k.getDebt(ctx, addr, poolId)
```

Comment:
I am not exactly sure how this is working. Debt key still have only user address, but ideally it should have address + pool id
This is keeping a single debt for a user even if the user opens multiple leverage lp positions. Older position gets lost when opening new one?